testkit/javadsl/src/main/scala/com/lightbend/lagom/javadsl/testkit/ServiceTest.scala

```diff
@@ -289,6 +328,13 @@ object ServiceTest {
     val port = Promise[Int]()
     val testServiceLocatorPort = TestServiceLocatorPort(port.future)
 
+    val sslPort: Option[Int] =
+      if (setup.ssl) {
+        Some(Random.nextInt(10000) + 5000) // A random value in the range [5000,15000)
```

Comment:
Why random instead of using `0` and later get the assigned port (like we have for HTTP)?

Reply:
Hmmm, I was using the `play.server.https.port` to setup the port and enable SSL, but I can do it programmatically. Good call.